toto/runlib.py

```diff
@@ -35,53 +35,63 @@
 else:
   import subprocess
 
-# XXX LP: I think we'll get rid of toto formats because we want to use
-# model validators instead for toto schemas, we'll still use
-# toto.ssl_crypto.formats though
-
-# import toto.formats
-
 import toto.models.link
 import toto.log as log
 
 import toto.ssl_crypto.hash
 import toto.ssl_crypto.formats
 
+def _hash_artifact(filepath, hash_algorithms=['sha256']):
+  """Internal helper that takes a filename and hashes the respective file's
+  contents using the passed hash_algorithms and returns a hashdict conformant
+  with ssl_crypto.formats.HASHDICT_SCHEMA. """
+  toto.ssl_crypto.formats.HASHALGORITHMS_SCHEMA.check_match(hash_algorithms)
+  hash_dict = {}
 
-def record_artifacts_as_dict(artifacts):
-  """Takes a list of directories and/or filenames as input and creates a dict
-  with filepaths as keys and their files' hashes as values.
+  for algorithm in hash_algorithms:
+    digest_object = toto.ssl_crypto.hash.digest_filename(filepath, algorithm)
+    hash_dict.update({algorithm: digest_object.hexdigest()})
```

Comment:
I admit it, I'm guilty of copy and paste. Maybe we should ask @vladimir-v-diaz [about this](https://github.com/theupdateframework/tuf/blame/develop/tuf/util.py#L410-L412)?